File: src/backend/note-server/http-server/src/main/java/lastpunch/notehttpserver/entity/Note.java
package lastpunch.notehttpserver.entity;

import java.time.LocalDateTime;
import java.util.List;
import javax.validation.constraints.NotNull;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import org.bson.types.ObjectId;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document("notes")
@Getter
@Setter
@Builder
@ToString
public class Note {
    @Id
    private ObjectId id;
    
    @NotNull
    private Long workspaceId;
    
    @NotNull
    private Long channelId;
    
    @NotNull
    private Long creatorId;
    
    private String title;
    private String content; // JSON string
    private List<Op> ops;
    private LocalDateTime createDt;
    private LocalDateTime modifyDt;
}

File: src/frontend/ios/Snack/Models/Entity/Data/Note.swift
//
//  Note.swift
//  Snack
//
//  Created by ghyeongkim-MN on 2022/02/08.
//

import Foundation

struct Note: ModelType {
    var id: String
    let title: String
    var creatorId: Int
    let createDt: String
    let modifyDt: String
}
